internal/activities/validate_file_formats.go

```diff
@@ -96,3 +86,34 @@ func (a *ValidateFileFormats) Execute(
 
 	return &ValidateFileFormatsResult{Failures: failures}, nil
 }
+
+func parseFormatList(r io.Reader) (formatList, error) {
+	var i int
+	formats := make(formatList)
+
+	cr := csv.NewReader(r)
+	for {
+		row, err := cr.Read()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return nil, fmt.Errorf("invalid CSV: %v", err)
+		}
+
+		// Skip the first row.
+		if i > 0 {
+			// Get the file format identifier from the first column of each row
+			// and ignore subsequent columns.
+			formats[row[0]] = struct{}{}
```

Comment:
Should we check for empty rows/columns?

Reply:
@jraddaoui I added a test to confirm that an empty row in an allowlist is ignored and doesn't cause an error.

I also added an explicit check for a "PRONOM PUID" (any letter case) header to allow for other columns in the allowlist, and added an extra column to the sample and test allowlist CSVs.

Reply:
Thanks!